libs/native/src/main21/java/org/elasticsearch/nativeaccess/jdk/JdkSystemdLibrary.java

```diff
@@ -40,10 +44,23 @@ static String findLibSystemd() {
                 continue;
             }
             try (var stream = Files.walk(basepath)) {
+                var foundpaths = stream.filter(Files::isDirectory)
+                    .map(p -> p.resolve(libsystemd))
+                    .filter(Files::exists)
+                    .map(p -> p.toAbsolutePath().toString())
+                    .toList();
 
-                var foundpath = stream.filter(Files::isDirectory).map(p -> p.resolve(libsystemd)).filter(Files::exists).findAny();
-                if (foundpath.isPresent()) {
-                    return foundpath.get().toAbsolutePath().toString();
+                UnsatisfiedLinkError last = null;
+                for (String path : foundpaths) {
+                    try {
+                        System.load(path);
```

Comment:
A QQ to see if I got this correctly: in multiarch you may have multiple libraries in the library path, for different archs (32/64). Depending on which arch the JVM is running (64bit I assume, unless something really strange, but your solution should work either way), loading the lib for a different arch will fail with a UnsatisfiedLinkError.
So you load every library you find, the first one that succeeds will be the right arch. If we are not able to load any, throw as before. Correct?

Reply:
Correct :)